cobbler.spec

```
@@ -384,6 +384,40 @@ chgrp %{apache_group} %{_sysconfdir}/cobbler/settings.yaml
 %else
 %post
 %systemd_post cobblerd.service
+
+if [ $1 -gt 1 ]; then
+    if [ -f /etc/susemanager-release ] || [ -f /etc/uyuni-release ]; then
+        if [ -f /etc/cobbler/settings.yaml.rpmnew ] && [ ! -f /etc/cobbler/settings.d/zz-uyuni.settings ]; then
+            # The /etc/cobbler/settings.yaml.backup should be existing unless explicitely removed.
+            if [ ! -f /etc/cobbler/settings.yaml.backup ]; then
+                echo "Not found: /etc/cobbler/settings.yaml.backup - Skipping settings migration"
+            else
+                python3 <<EOF
+# Check changes between old defaults (backup) and current settings
+# to detect local changes applied to settings. Local changes are then
+# moved to /etc/cobbler/settings.d/zz-uyuni.settings
+import yaml
+old_backup_settings = None
+current_settings = None
+new_uyuni_settings = {}
+with open("/etc/cobbler/settings.yaml.backup") as _input:
+    old_backup_settings = yaml.safe_load(_input)
+with open("/etc/cobbler/settings.yaml") as _input:
+    current_settings = yaml.safe_load(_input)
+for conf in current_settings:
+    if current_settings[conf] != old_backup_settings[conf]:
```

This could crash when a user defines a totally new property (e.g. when they make a typo in defining a property).

WDYT about something like this:

```suggestion
for current_key, current_val in current_settings.items():
    if old_backup_settings.get(current_key, "") != current_val:
```

NOTE: if you accept this suggestion, remember to fix the renamed `conf` -> `current_key` and use `current_val` .

---

Good catch! I'm fixing it